Lib/test/test_tempfile.py

```diff
@@ -39,27 +40,46 @@
 class TestLowLevelInternals(unittest.TestCase):
     def test_infer_return_type_singles(self):
         self.assertIs(str, tempfile._infer_return_type(''))
+        self.assertIs(str, tempfile._infer_return_type(pathlib.Path('')))
```

Comment:
It wouldn't hurt to add a test for a path-like object that returns bytes (you will have to make a test object yourself).